filter.h

```diff
@@ -0,0 +1,176 @@
+#ifndef _FILTER_H_
+#define _FILTER_H_
+
+//Standard Includes 
+#include <stdint.h>
+
+
+
+/** @file
+ * @defgroup MovingAvg Filter Module   
+ * 	@ingroup Filters
+ * 
+ * 
+ * This module implements a Filter using a linear transfer function that takes in scale/offset in order to calculate new values  
+ * new_value = raw_value * scale + offset 
+ * 
+ *
+ * This module defaults to a MovingAvg Filter function. The user is resposible for implementing a filter based on application. 
+ * It can be changed using the Set_Filter function 
+ * 
+ * 
+ * 
+ * 
+ * 
+ * Example for implementation with ADC module -- in progress 
+ * @code
+ * 
+ * 
+ * ADC_Init();
+ * Filter_Init();
+ * 
+ * QueueMeasurement()
+ * 
+ * 
+ * 
+ * 
+ * 
+ * 
+ * 
+ * 
+ * 
+ * @endcode
+ * 
+ * 
+ * 
+ * @author Neil Surti
+ * @author Alexander Revolous 
+ * 
+ * 
+ * **/ 
+
+
+
+
+
+
+
+
+/**
+ *  Filter_Init initliazes all variables/arrays for the filter i.e (min,max,index,) including setting the call back functions to NULL 
+ *  sets the scale/offset values from user and calculates the value for the shift  
```

Comment:
Documenting the details on how the implementation will be done (i.e. shift) should be in the .c file. Isolate the module users from such complexity. This applies to the documentation for Filter_Update as well. Again, documenting the implementation details is good but it belongs with the implementation in the .c file.